clients/p4rt-ctl/p4rt-ctl.in

```diff
@@ -769,6 +791,54 @@ def p4ctl_show(client, bridge):
 def p4ctl_set_pipe(client, bridge, device_config, p4info):
     client.set_fwd_pipe_config(p4info, device_config)
 
+TUNSETIFF = 0x400454CA
+IFF_TAP = 0x0002
+IFF_NO_PI = 0x1000
+MAX_FRAME_SIZE = 1500
+SELECT_TIMEOUT = 0.1
+
+class TapDevice:
+    def __init__(self, tap_name):
+        try:
+            self.tap = os.open("/dev/net/tun", os.O_RDWR)
+
+            # Configure the TAP interface.
+            ifr = struct.pack('16sH', tap_name.encode(), IFF_TAP | IFF_NO_PI)
+            fcntl.ioctl(self.tap, TUNSETIFF, ifr)
+
+            # Bring the TAP interface up.
+            os.system(f"sudo ip link set dev {tap_name} up")
+
+            print(f"Created TAP device {tap_name}")
+        except Exception as e:
+            print(f"Error creating TAP interface: {e}")
+
+    # Continuously reads pkts from TAP port and sends them out to the server
+    def read(self, client):
+        while True:
+            readable, _, _ = select.select([self.tap], [], [], SELECT_TIMEOUT)
+            if self.tap in readable:
+                # read pkt from tap port
+                data = os.read(self.tap, MAX_FRAME_SIZE)
+
+                # send the packet to the server
+                client.send_packet_out(data)
+
+    # Sends pkts to the tap port
+    def write(self, data):
+        os.write(self.tap, data)
+
+# Thread function to read pkts from TAP port
+def read_from_tap(tap, client):
+    print("Tx thread polling on TAP port")
+    tap.read(client)
+
+@with_client
+def p4ctl_start_pktio(client, bridge):
+    tap_device = TapDevice("pktioTap0")
```

Comment:
What happens if the TAP interface with this name already exists? Would it fall into the `except` section of the TapDevice class' `__init__`? 
Not sure if that is a good UX to have an exception log

Reply:
`os.open("/dev/net/tun", os.O_RDWR) ` creates an interface if it doesn't exist. If an interface with the same name already exists, then it is opened for read and write. No error is thrown in this case.